External page components/Public links preview.js

```diff
@@ -96,58 +98,83 @@ PublicLinkOverview.prototype = {
-                                    croppingType = publicLinkData["properties"]["ConversionConfiguration"]["cropping_configuration"]["cropping_type"];
-                                    width = publicLinkData["properties"]["ConversionConfiguration"]["width"];
-                                    height = publicLinkData["properties"]["ConversionConfiguration"]["height"];
-
-                                    // depending on the way of cropping, the width and height are stored in different objects
-                                    if (!width || width == null) {
-                                        width = publicLinkData["properties"]["ConversionConfiguration"]["cropping_configuration"]["width"];
-                                        height = publicLinkData["properties"]["ConversionConfiguration"]["cropping_configuration"]["height"];
-                                    }
-                                }
-
-                                if (croppingType === "Entropy") {
-                                    croppingType = "Smart crop";
-                                } else if (croppingType === "Custom") {
-                                    croppingType = "Custom crop";
-                                } else if (croppingType === "CentralFocalPoint") {
-                                    croppingType = "Crop to center";
-                                } else {
-                                    croppingType = "Uncropped";
-                                }
-
-                                var context = {title: title, href: publicLink, preview: thumbnail, rendition: rendition, width: width, height: height, croppingType: croppingType};
-                                var html = listItemTemplate(context);
-                                document.getElementById("publicLinkList").innerHTML += html;
-                            }
-                        });
-
+                    if (publicLinkAssetHref && publicLinkAssetHref !== "") {
+                        AJAX.push($.getJSON(publicLinkAssetHref));
                     }
                 }
+
+                $.when.apply($, AJAX).done(function () {
+                    // This callback will be called with multiple arguments,
+                    // one for each AJAX call
+                    // Each argument is an array with the following structure: [data, statusText, jqXHR]
+
+                    // Let's map the arguments into an object, for ease of use
+                    var responses = [];
+                    for (var i = 0, len = arguments.length; i < len; i++) {
+                        responses.push(arguments[i][0]);
+                    }
+
+                    responses = responses.sort(compareResponse);
+                    responses.forEach(element => {
+                        let html = convertToHtml(element);
```

Comment:
convertToHtml does not have a return value, so html is _undefined_ . It would be enough to just call convertToHtml as this method already adds the HTML to the desired element. You can then also remove the next line.